extensions/frontend-tiles/src/MeshExportServiceProvider.ts

```diff
@@ -0,0 +1,190 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
All functionality for the mesh export service itself (not dealing with batched tiles) has been moved to a separate file, and the class `MeshExportServiceProvider` has been created to house those functions. `obtainMeshExportTilesetUrl` is the main access point to the MES, so it has been left public, while the other functions have been made private.